### DISCUSSION TITLE
Pitch: Transition to Web Components

### DISCUSSION THREAD

**Discussion by ayo-run** [ayoayco/astro-reactive-library#270] [General]
Hi everyone!

For the past several months, it has been exciting creating and experimenting with Astro Components, and I would say we have had some level of success in creating a dynamic HTML Form builder that allows easy setting up of validation that runs on the server (render time) and on the browser (on DOM events). 

Recently I've been assessing the best way to move the project forward and I would like to start the discussion here about making the library more generic.

Basically, I think the code we have right now could be ported to use web components instead of Astro components, and still support integration with Astro with a few trade offs.

I will try to write more details soon and possibly start a new repo. Let me know what you think! 🙏

**Comment by fazzaamiarso** [comment]
@ayoayco I was thinking on migrating components to WC too when working on this library. I'm glad we're on same page. 😀
